packages/aws-rfdk/lib/deadline/lib/repository.ts

```diff
@@ -584,6 +585,10 @@ export class Repository extends Construct implements IRepository {
 
     this.version = props.version;
 
+    if ((props.secretsManagementSettings?.enabled ?? true) && this.version.isLessThan(Version.MINIMUM_SECRETS_MANAGEMENT_VERSION)) {
+      throw new Error(`The supplied Deadline version (${props.version.versionString}) does not support Deadline Secrets Management in RFDK. Either upgrade Deadline to the minimum required version (${Version.MINIMUM_SECRETS_MANAGEMENT_VERSION.versionString}) or disable the feature in the Repository's construct properties.`);
+    }
```

Comment:
I know I said we should throw an error like this if the version doesn't meet the requirement, but we discussed it on Slack yesterday and we landed on disabling the feature instead. We shouldn't have to change anything in the RenderQueue, but let's adjust this to something like:
```suggestion
    const meetsMinSecretsVersion = !this.version.isLessThan(Version.MINIMUM_SECRETS_MANAGEMENT_VERSION);
    const secretsManagementIsEnabled = props.secretsManagementSettings?.enabled ?? meetsMinSecretsVersion;
    
    if (secretsManamentIsEnabled && !meetsMinSecretsVersion) {
      throw new Error(`The supplied Deadline version (${props.version.versionString}) does not support Deadline Secrets Management in RFDK. Either upgrade Deadline to the minimum required version (${Version.MINIMUM_SECRETS_MANAGEMENT_VERSION.versionString}) or disable the feature in the Repository's construct properties.`);
    }
```

Reply:
Good point. Updated.